quinn/src/endpoint.rs

```diff
@@ -232,12 +227,13 @@ impl EndpointInner {
     }
 
     fn drive_incoming(&mut self) {
-        while let Ok(Async::Ready(())) = self.incoming.poll_ready() {
-            if let Some((_, conn)) = self.buffered_incoming.pop_front() {
-                self.incoming.start_send(NewConnection::new(conn)).unwrap();
-                self.inner.accept();
-            } else {
-                break;
+        for i in (0..self.incoming.len()).rev() {
+            match self.incoming[i].poll() {
+                Ok(Async::Ready(())) | Err(_) if !self.incoming_live => {
+                    let _ = self.incoming.swap_remove_back(i);
+                }
+                // It's safe to poll an already dead connection
+                _ => {}
```

Comment:
There's a judgement call here on whether we should hide connections that die before the application sees them or not. Any thoughts?

Reply:
I guess I'd want to know though maybe if we have fine-grained information about the cause of death we could pass through based on that?

Reply:
The `ConnectionError` we have is quite detailed, but what cases would merit discarding them?

I think I favor the current behavior, which always exposes the fate of any connection that had a valid `Initial`, if nothing else because it would make it easier to notice if someone is doing something weird to your server.

Reply:
Sounds good to me.